feat: rule creation guidelines for sigmahq (#64)
SigmaHQ/sigma-specification: sigmahq/sigmahq_specification.md

- [Titles](#titles)
- [Status](#status)
- [References](#references)
- [Single Item Lists](#single-item-lists)

## Structure

The rules consist of a few required sections and several optional ones.

```yaml
title [required]
id [required]
related [optional]
   - id {rule-id}
      type {type-identifier}
status [required]
description [required]
references [required]
author [required]
date [required]
modified [optional]
tags [required]
logsource [required]
   category [optional]
   product [optional]
   service [optional]
   definition [optional]
   ...
detection
   {search-identifier} [optional]
      {string-list} [optional]
      {map-list} [optional]
      {field: value} [optional]
   ...
   condition
fields [optional]
falsepositives [required]
level [required]
```

## Filenames

All rule filename must follow the convention described in [Sigmahq_filename_rule.md](./Sigmahq_filename_rule.md)

## Titles

All SigmaHQ rules must titles must use title casing

Example:

```yml
title: Suspicious Office Child Process
```

## Status

All newly created rules must start with a status of `experimental`

## References

All rules must provide a public reference when possible

> ## Single Item Lists

Reviewer: ```suggestion
## Detection

### Item Lists
```

Single item list must be expressed in the same line instead of multi-line.

Example of single list items:

```yml
detection:
    selection:
        Image|endswith: '\example.exe'
```

Example of multi item list:

```yml
detection:
    selection:
        Image|endswith:
            - '\example_1.exe'
            - '\example_2.exe'
            - '\example_3.exe'

> ```

Reviewer: ```suggestion
`

### Selection name

to keep it easy to read, use for the simple rules the name 'selection' and 'filter'
`yml
selection:
filter:
condition: selection and not filter
`
```

Can not put the 3 `

> # SigmaHQ Custom Specification <!-- omit in toc -->

Reviewer 2: ```suggestion
# SigmaHQ Rule Conventions <!-- omit in toc -->
```

> This document describe an addtional set of custom requirements enfroced by the SigmaHQ rule repository in order to ensure an easy to maintain rule base. For the general Sigma specficiation please read the [Sigma_specification.md](../Sigma_specification.md).

Reviewer 2: ```suggestion
This document describes an additional set of custom requirements enforced by the SigmaHQ rule repository in order to ensure an easy to maintain rule base. For the general Sigma specification please read the [Sigma_specification.md](../Sigma_specification.md).
```

> This document describes an additional set of custom requirements enforced by the SigmaHQ rule repository in order to ensure an easy to maintain rule base. For the general Sigma specification please read the [Sigma_specification.md](../Sigma_specification.md).

Reviewer 2: ```suggestion
This document describes an additional set of rule conventions enforced by the SigmaHQ rule repository in order to ensure an easy to maintain rule base. For the general Sigma specification please read the [Sigma_specification.md](../Sigma_specification.md).
```

Reviewer 2: If accepted, the file name should change as well

## Summary

- [Summary](#summary)
- [Structure](#structure)
- [Filenames](#filenames)
- [Titles](#titles)
- [Status](#status)
- [References](#references)
- [Single Item Lists](#single-item-lists)

## Structure

The rules consist of a few required sections and several optional ones.

```yaml
title [required]
id [required]
related [optional]
   - id {rule-id}
      type {type-identifier}
status [required]
description [required]
references [required]
author [required]
date [required]
modified [optional]
tags [required]
logsource [required]
   category [optional]
   product [optional]
   service [optional]
   definition [optional]
   ...
detection
   {search-identifier} [optional]
      {string-list} [optional]
      {map-list} [optional]
      {field: value} [optional]
   ...
   condition
fields [optional]
falsepositives [required]
level [required]
```

## Filenames

All rule filename must follow the convention described in [Sigmahq_filename_rule.md](./Sigmahq_filename_rule.md)

## Titles

> All SigmaHQ rules must titles must use title casing

Reviewer 2: ```suggestion
All SigmaHQ rule titles must use title casing
```
